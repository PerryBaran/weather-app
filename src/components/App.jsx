import React, { useEffect, useState } from "react";
import "../styles/App.css";
import LocationDetails from "./LocationDetails";
import ForecastSummaries from "./ForecastSummaries";
import ForecastDetails from "./ForecastDetails";
import getForecast from "../requests/getForecast";

function App() {
  const [{ forecasts, location }, setForecastData] = useState({
    location: {},
    forecasts: [],
  });
  const [selectedDate, setSelectedDate] = useState(0);

  const selectedForecast = forecasts.find(
    (forecast) => forecast.date === selectedDate
  );

  const handleForecaseSelect = (date) => {
    setSelectedDate(date);
  };

  useEffect(() => {
    getForecast(setForecastData, setSelectedDate);
  }, []);

  return (
    <div className="weather-app">
      <LocationDetails location={location} />
      <ForecastSummaries
        forecasts={forecasts}
        handleForecastSelect={handleForecaseSelect}
      />
      {selectedForecast && <ForecastDetails forecast={selectedForecast} />}
    </div>
  );
}

export default App;
