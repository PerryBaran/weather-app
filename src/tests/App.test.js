import { render, screen } from "@testing-library/react";
import React from "react";
import App from "../components/App";
import forecast from "../data/forecast.json";

test("renders learn react link", () => {
  render(<App location={forecast.location} />);
  const h1Element = screen.getByText(/Manchester, UK/i);
  expect(h1Element).toBeInTheDocument();
});
