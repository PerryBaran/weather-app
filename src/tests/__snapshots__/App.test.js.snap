// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app snapshot renders correctly 1`] = `
<DocumentFragment>
  <div
    class="weather-app"
  >
    <h1>
      , 
    </h1>
    <div
      class="forecast-summaries"
    />
  </div>
</DocumentFragment>
`;
